text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,458 @@
+export function getIsolationScope(): Scope;
+
+// Create a new isolation scope for this scope
+// This will NOT make this scope the isolation scope, but will create a new isolation scope (based on the active isolation scope, if one exists)
+scope.isolate();
+
+// Similar to `withScope`, but it forks a new scope AND sets a new isolation scope for this context
+export function withIsolationScope(callback: (scope) => void): void;
+```
+
+You can fetch the current isolation scope via `getIsolationScope()`. You can define a new isolation scope via `scope.isolate()`, which will define a new isolation scope for this scope, and for all scopes that will be forked off this scope. When a client is created & bound, an isolation scope will immediately be created, similar to the global scope for a client.
+
+An isolation scope is attached to the current execution context, similar to the current scope. There is always exactly one current isolation scope. If you call `getIsolationScope()` before a client has been created, a noop isolation scope is returned, which should be merged in once a client is actually created (same as with the global scope).
+
+Similar to the global scope, an isolation scope is always a separate scope, so nothing will inherit off it - except for a potential superseding isolation scope.
+If an isolation scope is created, and there is already an isolation scope in the current execution context, then the new isolation scope should be forked off the previous one (with copy-on-write):
+
+```js
+Sentry.init();
+
+// There is always an isolation scope
+const initialIsolationScope = Sentry.getIsolationScope();
+initialIsolationScope.setTag("tag1", "aa");
+
+// Auto-instrumentation automatically forks an execution context with a new isolation scope
+app.get("/my-route", function () {
+  const isolationScope = Sentry.getIsolationScope();
+  isolationScope !== initialIsolationScope;
+  // Isolation scope inherits from previously current isolation scope
+  isolationScope.getTags(); // -> tag1: aa
+});
+```
+
+Note that in environments where you do not have multiple execution contexts (e.g. Browser JS), there is basically only a single isolation scope that is fundamentally the same as the global scope. In these environments, global & isolation scope can be used completely interchangeably. However, for consistency we still have these concepts everywhere - but users do not need to care about them, and we can reflect this in the SDK specific documentation.
+
+### How to document isolation scopes
+
+Isolation scopes will be documented in a new section under "Enriching Events > Per request data" (or similar, based on SDK language & needs), which will explain how to use `getIsolationScope()` and when/how data added there will be applied.
+
+Global methods to set data, like `Sentry.setTag()` or `Sentry.setUser()`, will set on the isolation scope. Only if you want to set on a different scope, you need to manually get it (e.g. `getGlobalScope().setTag()`).
```

Comment:
On mobile, global methods should change the global scope, as we don't need an isolation scope, IMO.

Reply:
it can/should still set on the isolation scope, which in absence of execution contexts will be functionally basically global. But _if_ execution contexts are ever added in some way, this will then also "just work" - that's the idea here, at least.